### PR TITLE
Added Dev Namespace Creation Pipeline

### PR DESCRIPTION
**Pull request by bjcmit**
Added new AZDO pipeline that creates a new branch in the manifest repo containing the YAML files to create a new namespace

### CODE REVIEW

File: .pipelines/create-namespace.yaml [hunk 1]
```diff
@@ -0,0 +1,87 @@
...
+  vmImage: ubuntu-latest
+
+variables:
+ - group: az-vote-app-dev
+ - name: MANIFEST_DIR
+   value: $(System.DefaultWorkingDirectory)/$(MANIFESTS_REPO)
+ - name: NAMESPACE_NAME
+   value: ${{ parameters.namespace_name }}
+ - name: BRANCH_NAME
+   value: brysmith/namespace_creation
```
Comment (sudivate) [inline .pipelines/create-namespace.yaml:18]: dev branch

File: .pipelines/delete-namespace.yaml [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+   echo "Delete local branch $NAMESPACE_NAME"   
+   git branch -d $NAMESPACE_NAME
+   echo "Delete remote branch $NAMESPACE_NAME" 
+   git push origin --delete $NAMESPACE_NAME
+  displayName: Delete Branch
+  env: 
+    NAMESPACE_NAME: ${{ parameters.namespace_name }}
+
+- script: |
+   git checkout $BRANCH_NAME
```
Comment (sudivate) [inline .pipelines/delete-namespace.yaml:48]: Deleting of sync files should go before deletion of branch

File: .pipelines/create-namespace.yaml [hunk 3]
```diff
@@ -0,0 +1,87 @@
...
+
+resources:
+  repositories:
+  - repository: Manifest
+    type: git # Set to 'github' for a GitHub repository 
+    name: azure-vote-app-deployment
+    ref: 'refs/heads/brysmith/namespace_creation'
```
Comment (sudivate) [inline .pipelines/create-namespace.yaml:25]: dev branch ?

File: .pipelines/create-namespace.yaml [hunk 4]
```diff
@@ -0,0 +1,87 @@
...
+
+- script: git config --global user.email $PR_USER_EMAIL & git config --global user.name $PR_USER_NAME
+  displayName: Configure Git
+  env:
+    PR_USER_NAME: "Git Ops"
+    PR_USER_EMAIL: "agent@gitops.com"
+  workingDirectory: $(MANIFEST_DIR)
+
+- script: |
+   sudo wget https://github.com/mikefarah/yq/releases/download/v4.9.3/yq_linux_amd64 -O /usr/bin/yq
```
Comment (sudivate) [inline .pipelines/create-namespace.yaml:39]: Install yq from package manger

File: .pipelines/create-namespace.yaml [hunk 5]
```diff
@@ -0,0 +1,87 @@
...
+   sudo chmod +x /usr/bin/yq
+  displayName: Install yq
+
+- script: |
+   echo "Create a new branch $NAMESPACE_NAME"
```
Comment (sudivate) [inline .pipelines/create-namespace.yaml:44]: newly created branch should only have low level manifest (.yaml) of k8 native objects

File: .pipelines/delete-namespace.yaml [hunk 6]
```diff
@@ -0,0 +1,59 @@
...
+  vmImage: ubuntu-latest
+
+variables:
+ - group: az-vote-app-dev
+ - name: MANIFEST_DIR
+   value: $(System.DefaultWorkingDirectory)/$(MANIFESTS_REPO)
+ - name: NAMESPACE_NAME
+   value: ${{ parameters.namespace_name }}
+ - name: BRANCH_NAME
+   value: brysmith/namespace_creation
```
Comment (sudivate) [inline .pipelines/delete-namespace.yaml:18]: dev branch

File: .pipelines/delete-namespace.yaml [hunk 7]
```diff
@@ -0,0 +1,59 @@
...
+
+resources:
+  repositories:
+  - repository: Manifest
+    type: git # Set to 'github' for a GitHub repository 
+    name: azure-vote-app-deployment
+    ref: 'refs/heads/brysmith/namespace_creation'
```
Comment (sudivate) [inline .pipelines/delete-namespace.yaml:25]: dev branch

File: .pipelines/create-namespace.yaml [hunk 8]
```diff
@@ -0,0 +1,87 @@
...
+   echo $commitid
+   git push --set-upstream origin $BRANCH_NAME
+   echo "##vso[task.setvariable variable=COMMIT_ID]$commitid"
+   echo "##vso[task.setvariable variable=foo;isOutput=true]bar"
+  displayName: Add New Namespace Sync
+  env: 
+    NAMESPACE_NAME: ${{ parameters.namespace_name }}
+
+- script: |
+    git checkout $BRANCH_NAME
```
Comment (sudivate) [inline .pipelines/create-namespace.yaml:81]: can you move this script to kaizentm/cloud-native-ops/utils ?

File: .pipelines/create-namespace.yaml (hunk 1 above)
Comment (bjcmit) [reply .pipelines/create-namespace.yaml:18]: Changed

File: .pipelines/delete-namespace.yaml (hunk 2 above)
Comment (bjcmit) [reply .pipelines/delete-namespace.yaml:48]: Changed ordering

File: .pipelines/create-namespace.yaml (hunk 3 above)
Comment (bjcmit) [reply .pipelines/create-namespace.yaml:25]: Changed

File: .pipelines/create-namespace.yaml (hunk 4 above)
Comment (bjcmit) [reply .pipelines/create-namespace.yaml:39]: yq from the package manager is 3.3.2 https://launchpad.net/~rmescandon/+archive/ubuntu/yq

File: .pipelines/delete-namespace.yaml (hunk 6 above)
Comment (bjcmit) [reply .pipelines/delete-namespace.yaml:18]: Changed

File: .pipelines/delete-namespace.yaml (hunk 7 above)
Comment (bjcmit) [reply .pipelines/delete-namespace.yaml:25]: Changed

File: .pipelines/create-namespace.yaml (hunk 8 above)
Comment (bjcmit) [reply .pipelines/create-namespace.yaml:81]: Moved

File: .pipelines/create-namespace.yaml (hunk 5 above)
Comment (bjcmit) [reply .pipelines/create-namespace.yaml:44]: So what exactly needs to be removed?

Comment (sudivate) [reply .pipelines/create-namespace.yaml:44]: As per current template , the branch should have three yaml files , namespace, role and role binding

File: .pipelines/create-namespace.yaml (hunk 4 above)
Comment (bjcmit) [reply .pipelines/create-namespace.yaml]: Changed to use snap

File: utils/get-commit-status.sh [hunk 9]
```diff
@@ -0,0 +1,33 @@
...
+echo $REPO
+echo $COMMIT_ID
+echo "end of list"
+
+set -euo pipefail  # fail on error
+
+export SYNC_INTERVAL=5
+#SYSTEM_ACCESSTOKEN=$(printf "%s"":$PAT" | base64)
+STATUS_UPDATE=false
+while [ "$STATUS_UPDATE" = false ]; do
```
Comment (sudivate) [inline utils/get-commit-status.sh:20]: Please update the while loop to time out like [here](https://github.com/kaizentm/cloud-native-ops/blob/638ff710d55cd725bd4cb59166d324f5f9465de8/utils/delete-namespace.sh#L30)

**Review by sudivate — APPROVED** [review]
Please update the while loop to time out before merge

File: utils/get-commit-status.sh (hunk 9 above)
Comment (bjcmit) [reply utils/get-commit-status.sh:20]: Updated

**Pull request merged**